android/README.md

```diff
@@ -0,0 +1,226 @@
+| ---        | ---           |
+| Duration   | 5-10 minutes  | 
+| Difficulty | Medium        |
+
+1. All the app's interactions with okhttp are found in the `com.splunk.android.workshopapp.FirstFragment` class. Open up
+   that class in Android Studio.
+2. Find the `buildOkHttpClient()` method.
+3. Change the last line of the method to read:
+   ```
+   return SplunkRum.getInstance().createRumOkHttpCallFactory(builder.build());
+   ```
+   Remember to add an import for the `com.splunk.rum.SplunkRum` class when prompted by Android Studio.
+4. Build and restart the app. Now try generating some http traffic by using the 3 buttons on the first screen of the app
+   labeled "LOGIN", "HTTP NOT FOUND", and "HTTP ERROR".
+5. Navigate to the Splunk RUM page in the Splunk Observability product.
+    1. Filter for your app and environment.
+    2. Try clicking on the http metrics and finding example sessions for the results. (These metrics are "Endpoint
+       Errors" and "Endpoint Latency")
+
+## Part 4: Add some manual instrumentation
+
+The SplunkRum class in the Android RUM library contains a bunch of options for doing manual instrumentation and for
+customizing the telemetry that is sent to Splunk RUM. In this part of the workshop, you'll try a couple of them, and you
+can explore the rest on your own.
+
+| Objective  | To add some manually instrumented spans and events to the app. |
+| ---        | ---            |
+| Duration   | 10 minutes +   | 
+| Difficulty | Advanced       |
+
+#### 1. Add a "workflow" to the app, and add some custom attributes.
+
+In Splunk's RUM implementation, a "workflow" is a custom span for which metrics will be automatically collected. In the
+Android instrumentation, these workflows are modeled as OpenTelemetry Span instances.
+
+Let's create a workflow for the user's login. This will be triggered by the button labled "LOGIN" on the first screen of
+the app.
+
+1. The code for this is in the `com.splunk.android.workshopapp.FirstFragment` class. Open up that class in Android
+   Studio.
+2. Find this block of code in the `onViewCreated` method:
+   ```
+        binding.loginButton.setOnClickListener(v -> {
+            //not really a login, but it does make an http call
+            makeCall("https://ssidhu.o11ystore.com/");
+        });
+   ```
+3. Change this block of code to look like this:
+   ```
+        Span loginSpan = SplunkRum.getInstance().startWorkflow("Login");
+        try (Scope s = loginSpan.makeCurrent()) {
+            //not really a login, but it does make an http call
+            makeCall("https://ssidhu.o11ystore.com/");
+        } finally {
+            loginSpan.end();
+        }
+   ```
+   When prompted by Android Studio, add imports for the OpenTelemetry `Span` and `Scope` classes.
+4. Add a custom "global attribute" after the `makeCall` line (but before the `finally` block):
+   ```
+   SplunkRum.getInstance().setGlobalAttribute(stringKey("user_id"), "123456")
```

Comment:
```suggestion
   SplunkRum.getInstance().setGlobalAttribute(stringKey("user_id"), "123456");
```

Reply:
If we assume that the ones doing this workshop don't have much experience with java it might be better to use regular import instead of static import.

Reply:
I'll keep this in mind when running the workshop. If people get confused, I'll adjust it, but I'd like to see if they can grok it as-is first, since it is a good java feature that people should know.